edgelet/iotedge/src/support_bundle.rs

```diff
@@ -157,6 +165,134 @@ where
+            .args(&["-u", "iotedge"])
+            .args(&["-S", &since])
+            .arg("--no-pager")
+            .output();
+
+        #[cfg(windows)]
+        let inspect = ShellCommand::new("Get-WinEvent")
+            .args(&[
+                "-ea",
+                "SilentlyContinue",
+                "-FilterHashtable",
+                &format!(
+                    "@{{ProviderName='iotedged';LogName='application';StartTime={}}}",
+                    since
+                ),
+                "Select",
+                "TimeCreated",
+                "Message",
+                "Sort-Object",
+                "@{Expression='TimeCreated';Descending=$false}",
+                "Format-Table",
+                "-AutoSize",
+                "-Wrap",
+            ])
+            .output();
+
+        let (file_name, output) = if let Ok(result) = inspect {
+            if result.status.success() {
+                ("edgelet.txt", result.stdout)
+            } else {
+                ("edgelet_err.txt", result.stderr)
+            }
+        } else {
+            let err_message = inspect.err().unwrap().description().to_owned();
+            println!("Could not find system logs for iotedge. Including error in bundle.\nError message: {}", err_message);
+            ("edgelet_err.txt", err_message.as_bytes().to_vec())
+        };
+
+        state
+            .zip_writer
+            .start_file_from_path(&Path::new("logs").join(file_name), state.file_options)
+            .map_err(|err| Error::from(err.context(ErrorKind::WriteToFile)))?;
+
+        state
+            .zip_writer
+            .write(&output)
+            .map_err(|err| Error::from(err.context(ErrorKind::WriteToFile)))?;
+
+        Ok(state)
+    }
+
+    fn write_check_to_file(mut state: BundleState<M>) -> Result<BundleState<M>, Error> {
+        let iotedge = env::args().into_iter().nth(0).unwrap();
+        println!("Calling iotedge check");
+        let check = ShellCommand::new(iotedge)
+            .arg("check")
+            .args(&["-o", "json"])
+            .output()
+            .map_err(|err| Error::from(err.context(ErrorKind::BundleCheck)))?;
+
+        state
+            .zip_writer
+            .start_file_from_path(&Path::new("check.json"), state.file_options)
+            .map_err(|err| Error::from(err.context(ErrorKind::WriteToFile)))?;
+
+        state
+            .zip_writer
+            .write(&check.stdout)
+            .map_err(|err| Error::from(err.context(ErrorKind::WriteToFile)))?;
+
+        println!("Wrote check output to file");
+        Ok(state)
+    }
+
+    fn write_inspect_to_file(
+        mut state: BundleState<M>,
+        module_name: String,
+    ) -> Result<BundleState<M>, Error> {
+        println!("Running docker inspect for {}", module_name);
+        let mut inspect = ShellCommand::new("docker");
+
+        #[cfg(windows)]
+        inspect.args(&["-H", "npipe://./pipe/iotedge_moby_engine"]);
```

Comment:
This is only the case if the user is using Windows containers.

Admittedly it's hard at this layer to read the config.yaml and figure out what the real URI is. And it's likely that anyone using `iotedge support-bundle` is in a production scenario, so they're not using Linux containers.

So if you want to leave it like this, it's fine, but mention this justification in a comment.